src/averageAges.js

```diff
@@ -54,8 +83,29 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
-function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+function calculateAverageAgeDiff(people, onlyWithSon = false) {
+  let childCount = 0;
+
+  const childrenList = people.map(person => {
+    return (onlyWithSon === false && person.mother !== null) ? (
+      person.name
+    ) : (onlyWithSon && person.mother !== null && person.sex === 'm') ? (
+      person.name
+    ) : null;
+  });
```

Comment:
use filter to first find mothers
and them use filter to find all children (check for onlyWithSon inside this filter)